app/src/main/java/com/bogdan/codeforceswatcher/activity/AddUserActivity.kt

```diff
@@ -49,15 +49,22 @@ class AddUserActivity : AppCompatActivity(), OnClickListener {
                         val localUser = response.body()!!.result.firstOrNull()!!
                         ratingCall.enqueue(object : Callback<RatingChangeResponse> {
                             override fun onResponse(call: Call<RatingChangeResponse>, response: Response<RatingChangeResponse>) {
+                                progressBar.visibility = View.INVISIBLE
                                 if (response.isSuccessful) {
                                     localUser.ratingChanges = response.body()!!.result
-                                    CwApp.app.userDao.insert(localUser)
-                                    progressBar.visibility = View.INVISIBLE
+                                    if (CwApp.app.userDao.getAll().find { it.handle == localUser.handle } != null) {
```

Comment:
Move `CwApp.app.userDao.getAll().find { it.handle == localUser.handle }` to boolean variable (think about name, not use just `a`). Swap `!=`.